### PR TITLE
Add constant comparison operator to jq evaluation

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

This adds a `constant` field to the `jq` evaluation engine.
This allows us to write ruletypes like the one below, which compares the evaluation result to a constant value, rather than requiring the profile to supply to value.

The example below compares the field `delete_branch_on_merge` in the evaluation result to the constant `true`.
```
def:
  in_entity: repository
  rule_schema:
    type: object
    properties: {}
  ingest:
    type: rest
    rest:
      endpoint: "/repos/{{.Entity.Owner}}/{{.Entity.Name}}"
      parse: json
  eval:
    type: jq
    jq:
    - ingested:
        def: '.delete_branch_on_merge'
      constant: true
```

This also changes the `jq` equality check to not differentiate between number type (`int`, `float64` etc).
The alternative would be to let the user specify the type of their number constant, but I'm making the assumption that the likeliness of someone needing that is small, and not worth the additional overhead of setting a type every time.

Ref https://github.com/stacklok/minder-rules-and-profiles/issues/156

I will update the docs once this is approved.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/eval/jq/jq.go [hunk 1]
```diff
@@ -42,11 +42,11 @@ func NewJQEvaluator(assertions []*pb.RuleType_Definition_Eval_JQComparison) (*Ev
 
 	for idx := range assertions {
 		a := assertions[idx]
-		if a.Profile == nil {
-			return nil, fmt.Errorf("missing profile accessor")
+		if a.Profile == nil && a.Constant == nil {
+			return nil, fmt.Errorf("missing profile or constant accessor")
 		}
```
Comment (evankanderson) [inline internal/engine/eval/jq/jq.go:47]: Do you want `Profile` and `Constant` to be XOR (at least one set, but not both)?

File: internal/engine/eval/jq/jq.go [hunk 2]
```diff
@@ -88,9 +99,9 @@ func (jqe *Evaluator) Eval(ctx context.Context, pol map[string]any, res *engif.R
...
 
 		// Deep compare
-		if !reflect.DeepEqual(profileVal, dataVal) {
+		if !reflect.DeepEqual(numberToFloat64(expectedVal), numberToFloat64(dataVal)) {
```
Comment (evankanderson) [inline internal/engine/eval/jq/jq.go:102]: How does this work for booleans (like your example)?  I believe you that it does, but I was surprised to see that it worked.

Oh, I see!  `numberToFloat64` actually returns `any`, and passes through other values.  I'm trying to figure out if there's a better name -- maybe `standardizeNumbers`?

File: internal/engine/eval/jq/jq_test.go [hunk 3]
```diff
@@ -271,6 +317,27 @@ func TestValidJQEvals(t *testing.T) {
...
 			},
 		},
+		{
+			name: "valid single rule evaluates constant bool",
+			assertions: []*pb.RuleType_Definition_Eval_JQComparison{
+				{
+					Constant: &structpb.Value{
+						Kind: &structpb.Value_BoolValue{
+							BoolValue: false,
+						},
```
Comment (evankanderson) [inline internal/engine/eval/jq/jq_test.go:327]: I think you can write this as:

```suggestion
					Constant: &structpb.NewBoolValue(false),
```

File: internal/engine/eval/jq/jq_test.go [hunk 4]
```diff
@@ -292,6 +359,45 @@ func TestValidJQEvals(t *testing.T) {
...
+									},
+									{
+										Kind: &structpb.Value_StringValue{
+											StringValue: "c",
+										},
+									},
+								},
+							},
+						},
+					},
```
Comment (evankanderson) [inline internal/engine/eval/jq/jq_test.go:388]: I love that we're testing list input. 🎉 

Again, I think the helpers in `structpb` will make this less painful to read and write:

```suggestion
					Constant: &structpb.NewListValue(structpb.ListValue{Values: {
						structpb.NewStringValue("a"),
						structpb.NewStringValue("b"),
						structpb.NewStringValue("c"),
						},
					}}),
```

Unfortunately, `NewList` returns a `*ListValue, error`, so we can't just feed it into `NewListValue` directly.  (We could also do it _outside_ the function, or write an `errorToPanic[T](T, err) T` helper method.)

**Review by evankanderson — APPROVED** [review]
This makes a lot of sense, and it's sort of amazing that it was missing for so long.  Thanks for adding the test coverage, too!

File: internal/engine/eval/jq/jq.go (hunk 1 above)
Comment (eleftherias) [reply internal/engine/eval/jq/jq.go]: Good point, I've updated it

File: internal/engine/eval/jq/jq.go (hunk 2 above)
Comment (eleftherias) [reply internal/engine/eval/jq/jq.go]: Updated the name

**Review by JAORMX — APPROVED** [review]

**Pull request merged**